src/web/app/pages-instructor/instructor-track-view-page/instructor-track-view-page.component.ts

```diff
@@ -205,50 +210,38 @@ export class InstructorTrackViewPageComponent implements OnInit {
+                  .find((fsLog: FeedbackSessionLog) =>
+                    fsLog.feedbackSessionData.feedbackSessionName === this.formModel.feedbackSessionName);
 
-                logs.feedbackSessionLogs
-                  .filter((fsLog: FeedbackSessionLog) =>
-                    fsLog.feedbackSessionData.feedbackSessionName === this.formModel.feedbackSessionName)[0]
-                    .feedbackSessionLogEntries
+                if (!targetFeedbackSessionLog) {
+                  return;
+                }
+
+                targetFeedbackSessionLog.feedbackSessionLogEntries
                   .filter((entry: FeedbackSessionLogEntry) =>
                     LogType[entry.feedbackSessionLogType.toString() as keyof typeof LogType]
-                      === LogType.FEEDBACK_SESSION_VIEW)
-                  .map((entry: FeedbackSessionLogEntry) => {
-                    if (!(entry.studentData.email in this.studentToLog)
-                      || this.studentToLog[entry.studentData.email].timestamp < entry.timestamp) {
-                      this.studentToLog[entry.studentData.email] = entry;
-                    }
-                  });
-                this.searchResult = this.toFeedbackSessionLogModel(logs.feedbackSessionLogs[0]);
+                      === LogType.FEEDBACK_SESSION_VIEW
+                    && (!(entry.studentData.email in this.studentToLog)
+                        || this.studentToLog[entry.studentData.email].timestamp < entry.timestamp))
+                  .forEach((entry: FeedbackSessionLogEntry) => this.studentToLog[entry.studentData.email] = entry);
+
+                this.searchResult = this.toFeedbackSessionLogModel(targetFeedbackSessionLog);
               });
         }, (e: ErrorMessageOutput) => this.statusMessageService.showErrorToast(e.error.message));
   }
 
-  /**
-   * Return an empty search result for feedback sessions that is not published
-   */
-  // private emptyResult(): void {
-  //   this.searchResult = {
-  //     courseId: this.formModel.courseId,
-  //     feedbackSessionName: this.formModel.feedbackSessionName,
-  //     publishedDate: '',
-  //     logColumnsData: [],
-  //     logRowsData: [],
-  //   };
-  //   this.hasResult = true;
-  //   this.isSearching = false;
-  // }
-
-  private openModal(feedbackSession: FeedbackSession): void {
+  private openModal(): void {
     const modalContent: string = 'Published date of selected feedback session is more than 30 days ago. '
       + 'Only activities within the last 30 days will be shown.';
     const modalRef: NgbModalRef =
       this.simpleModalService.openConfirmationModal('Continue the operation?', SimpleModalType.WARNING, modalContent);
     modalRef.result.then(
       () => {
         const earliestSearchDate: Date =
-          new Date(Date.now() - this.LOGS_RETENTION_PERIOD * 24 * 60 * 60 * 1000 + 60 * 1000);
+          new Date(Date.now() - this.LOGS_RETENTION_PERIOD_IN_MILLISECONDS + 60 * 1000);
```

Comment:
I'm wondering if 60 * 1000 should be regarded as magic number here?
Instinctively when you read this line, you need to first figure out that the middle operand is "milliseconds", then try to use this to decipher `60 * 1000` which is "sixty second".
Just a discussion again. Not stating my feeling is right.

Reply:
I think it will be better if I declare 60*1000 as a constant `ONE_MINUTE_IN_MILLISECONDS`

Reply:
I guess this reads better as it stays consistent with the const before.